DSCResources/MSFT_NetworkTeam/MSFT_NetworkTeam.psm1

```diff
@@ -0,0 +1,243 @@
+    Param
+    (
+        [Parameter(Mandatory)]
+        [string]$name,
+
+        [Parameter(Mandatory)]
+        [String[]]$teamMembers
+    )
+    
+    $configuration = @{
+        name = $name
+        teamMembers = $teamMembers
+    }
+
+    Write-Verbose ($localizedData.GetTeamInfo -f $name)
+    $networkTeam = Get-NetLBFOTeam -Name $name -ErrorAction SilentlyContinue
+
+    if ($networkTeam) 
+    {
+        Write-Verbose ($localizedData.FoundTeam -f $name)
+        if ($null -eq (Compare-Object -ReferenceObject $teamMembers -DifferenceObject $networkTeam.Members))
+        {
+            Write-Verbose ($localizedData.teamMembersExist -f $name)
+            $configuration.Add('loadBalancingAlgorithm', $networkTeam.loadBalancingAlgorithm)
+            $configuration.Add('teamingMode', $networkTeam.teamingMode)
+            $configuration.Add('ensure','Present')
+        }
+    }
+    else
+    {
+        Write-Verbose ($localizedData.TeamNotFound -f $name)
+        $Configuration.Add('ensure','Absent')
+    }
+    $Configuration
+}
+
+Function Set-TargetResource 
+{
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory)]
+        [string]$name,
+
+        [Parameter(Mandatory)]
+        [String[]]$teamMembers,
+    
+        [Parameter()]
+        [ValidateSet("SwitchIndependent", "LACP", "Static")]
+        [String]$teamingMode = "SwitchIndependent",
+
+        [Parameter()]
+        [ValidateSet("Dynamic", "HyperVPort", "IPAddresses", "MacAddresses", "TransportPorts")]
+        [String]$loadBalancingAlgorithm = "HyperVPort",
+
+        [ValidateSet('Present', 'Absent')]
+        [String]$ensure = 'Present'
+    )
+    Write-Verbose ($localizedData.GetTeamInfo -f $name)
+    $networkTeam = Get-NetLBFOTeam -Name $name -ErrorAction SilentlyContinue
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($networkTeam)
+        {
+            Write-Verbose ($localizedData.foundTeam -f $name)
+            $setArguments = @{
+                'name' = $name
+            }
+
+            if ($networkTeam.loadBalancingAlgorithm -ne $loadBalancingAlgorithm)
+            {
+                Write-Verbose ($localizedData.lbAlgoDifferent -f $loadBalancingAlgorithm)
+                $SetArguments.Add('loadBalancingAlgorithm', $loadBalancingAlgorithm)
+                $isNetModifyRequired = $true
+            }
+
+            if ($networkTeam.TeamingMode -ne $teamingMode)
+            {
+                Write-Verbose ($localizedData.teamingModeDifferent -f $teamingMode)
+                $setArguments.Add('teamingMode', $teamingMode)
+                $isNetModifyRequired = $true
+            }
+            
+            if ($isNetModifyRequired)
+            {
+                Write-Verbose ($localizedData.modifyTeam -f $name)
+                Set-NetLbfoTeam @setArguments -ErrorAction Stop -Confirm:$false
+            }
+
+            $netTeamMembers = Compare-Object -ReferenceObject $teamMembers -DifferenceObject $networkTeam.Members
```

Comment:
I updated a few lines to go under 100 char limit. However, there are a few more than are over 100 but less than 110. I am leaving them as is for better readability.